Mathlib/Algebra/Order/BigOperators/Ring/Multiset.lean

```diff
@@ -13,10 +14,55 @@ This file contains the results concerning the interaction of multiset big operat
 rings.
 -/
 
+open Multiset
+
 @[simp]
 lemma CanonicallyOrderedAdd.multiset_prod_pos {R : Type*}
     [CommSemiring R] [PartialOrder R] [CanonicallyOrderedAdd R] [NoZeroDivisors R] [Nontrivial R]
     {m : Multiset R} : 0 < m.prod ↔ ∀ x ∈ m, 0 < x := by
   rcases m with ⟨l⟩
   rw [Multiset.quot_mk_to_coe'', Multiset.prod_coe]
   exact CanonicallyOrderedAdd.list_prod_pos
+
+section OrderedCommSemiring
+
+variable {α β : Type*} [CommMonoid α] [OrderedCommSemiring β]
+
+theorem map_multiset_prod_le_of_submultiplicative_of_nonneg {f : α → β} (h0 : ∀ a, 0 ≤ f a)
```

Comment:
I think you should just generalize `Multiset.le_prod_of_submultiplicative` to `MulPosMono` instead. I don't think this result has anything to do with rings.

Reply:
`MulPosMono` is not imported by `Algebra.Order.BigOperators.Group.Multiset`. 
What do you suggest the precise generalization should be, and to which file should it be moved? Or would it be ok to change the imports?
In my applications ` β` is always the real numbers. I could more generally imagine this being applied when `f` is an `AbsoluteValue`, in which case the codomain would also be a semiring. Do you have any applications in mind in which ` β`  is not a semiring?

Reply:
I don't think @erdOne's suggestion works: this is the ring-like lemma corresponding to the group-like lemma `Multiset.le_prod_of_submultiplicative`. Note: adding `MulPosMono` to the existing result is not a generalization: the current version doesn't require that the type has a 0.

Reply:
By generalize I meant generalize the proof and not derive it ad hoc. Not least because we are currently missing the instance (we should add it back) but I feel the proof will be more straightforward that way.

Reply:
I can do `[CommMonoidWithZero β] [PartialOrder β] [PosMulMono β]`. Both `α` and `β` be to be commutative for `Multiset.prod` to be defined.